src/main/java/com/aws/greengrass/device/ClientDevicesAuthService.java

```diff
@@ -155,7 +162,27 @@ protected void install() throws InterruptedException {
             Topics deviceGroupTopics = this.config.lookupTopics(CONFIGURATION_CONFIG_KEY, DEVICE_GROUPS_TOPICS);
             Topic caTypeTopic = this.config.lookup(CONFIGURATION_CONFIG_KEY, CA_TYPE_TOPIC);
 
-            if (whatHappened == WhatHappened.initialized) {
+            // Attempt to update the thread pool size as needed
+            try {
+                int threadPoolSize = Coerce.toInt(this.config.findOrDefault(DEFAULT_THREAD_POOL_SIZE,
+                        CONFIGURATION_CONFIG_KEY, THREAD_POOL_SIZE_TOPIC));
+                if (threadPoolSize >= cloudCallThreadPool.getCorePoolSize()) {
```

Comment:
That isn't allowed, it would only be for 0 and negative, which no one should be setting and expect to work. I don't see any need to log that; we will continue to operate normally.